app/models/hostgroup.rb

```diff
@@ -232,4 +235,8 @@ def validate_subnet_types
     errors.add(:subnet, _("must be of type Subnet::Ipv4.")) if self.subnet.present? && self.subnet.type != 'Subnet::Ipv4'
     errors.add(:subnet6, _("must be of type Subnet::Ipv6.")) if self.subnet6.present? && self.subnet6.type != 'Subnet::Ipv6'
   end
+
+  def suggest_default_pxe_loader
+    self.pxe_loader = self.try(:operatingsystem).try(:preferred_loader) unless self.pxe_loader.present?
```

Comment:
Oddly enough I see a 500 on this method when I visit Hostgroups#index. Even after the migration, when all hostgroups have 'PXELinux BIOS' set as 'pxe_loader'. 

It doesn't happen if I change `subtree_ids` to `subtree` on `children_hosts_count`, so I'm guessing it's a bug in Ancestry... `Host::Managed.authorized.where(:hostgroup => subtree).size` works but `Host::Managed.authorized.where(:hostgroup => subtree_ids).size` returns that error
```
 | ActionView::Template::Error (missing attribute: pxe_loader):
 |     21:           <%= link_to hosts_count.fetch(hostgroup.id, 0), hosts_path(:search => %Q[hostgroup_fullname = "#{hostgroup.title}"]) %>
 |     22:         </td>
 |     23:         <td>
 |     24:           <%= link_to hostgroup.children_hosts_count, hosts_path(:search => %Q[parent_hostgroup = "#{hostgroup.title}"]) %>
 |     25:         </td>
 |     26:         <td>
 ```